linux_docker_resources/entry_point.sh

```diff
@@ -22,13 +22,18 @@ echo "done."
 
 # We only attempt to install Connext on amd64
 if [ "${ARCH}" != "aarch64" ]; then
-    # extract all ignored rmws
-    # extract args between --ignore-rmw until the first appearance of '-'
-    IGNORE_CONNEXTDDS=`echo ${CI_ARGS} | sed -e 's/.*ignore-rmw \([^-]*\).*/\1/' | sed -e 's/-.*//' | grep rmw_connextdds`
-    IGNORE_CONNEXTCPP=`echo ${CI_ARGS} | sed -e 's/.*ignore-rmw \([^-]*\).*/\1/' | sed -e 's/-.*//' | grep rmw_connext_cpp`
-    # Install RTI Connext DDS if we didn't find both `rmw_connextdds` and `rmw_connext_cpp`
-    # within the "ignored RMWs" option string.
-    if [ -z "${IGNORE_CONNEXTDDS}" -o -z "${IGNORE_CONNEXTCPP}" ]; then
+    IGNORE_CONNEXTDDS=""
+    ignore_rwm_seen="false"
+    for arg in ${CI_ARGS} ; do
+        case $arg in
+            ("--ignore-rmw") ignore_rmw_seen="true" ;;
+            ("-"*) ignore_rmw_seen="false" ;;
```

Comment:
The refactor here is worth some attention and impacts my request to split the `--ignore-rmw` changes to other rmw into a separate PR. I still think I prefer it, even though it means including this refactor before dropping rmw_connext.

I haven't seen where in the above logic the closing `-` is added. Also, why a single `-` rather than the typical `--` to close an argument list? At first I thought perhaps an option parser was stripping the initial `-` but since `--ignore-rmw` is intact in the case above that doesn't seem the case.

Reply:
> I haven't seen where in the above logic the closing `-` is added. Also, why a single `-` rather than the typical `--` to close an argument list? At first I thought perhaps an option parser was stripping the initial `-` but since `--ignore-rmw` is intact in the case above that doesn't seem the case.

So to be clear, I'm not actually looking for a `--` to quit an option list here.  What I'm looking for is *any* other option that would stop us from processing this list (and options start with a single or double `-`).

That is, I'm trying to handle all of the following:

* `--option foo --ignore-rmw rmw_foo --ignore-rmw rmw_bar`
* `--option foo --ignore rmw rmw_foo rmw_bar`
* `--option foo --ignore-rmw rmw_foo --ignore-rmw rmw_bar --option1`
* `--option foo --ignore-rmw rmw_foo rmw_bar -- blah` (in this case, the `--` is required as otherwise it would be considered as part of the `--ignore-rmw` list)

All of that said, I'm going to actually recreate this comment in the new PR I'm about to open so we can continue to discuss it there.